pkg/operator/ceph/cluster/crash/reconcile.go

```diff
@@ -117,6 +118,12 @@ func (r *ReconcileNode) reconcile(request reconcile.Request) (reconcile.Result,
 			logger.Errorf("more than one CephCluster found in the namespace %q, choosing the first one %q", namespace, cephCluster.GetName())
 		}
 
+		clusterImage := cephCluster.Spec.CephVersion.Image
+		cephVersion, err := version.GetImageVersion(clusterImage)
```

Comment:
The latest approach looks good and just like @travisn said, a bit of polling wouldn't hurt in the case of the operator restarting.

Reply:
Agreed; I think the word "lock" above made me think "mutex" rather than "retry loop". Implementing a poll.

Reply:
The lock I'm thinking of would involve a simple mutex that is only used when updating the map or reading from the map, thus preventing any invalid access while the map is being updated.

Reply:
@travisn Ah, got it; Go 1.6 requires a lock for concurrent read and write. That explains the feedback, and why the book I studied didn't suggest the pattern in this patch would be a problem. Thank you. Updating.

(https://stackoverflow.com/questions/36167200/how-safe-are-golang-maps-for-concurrent-read-write-operations)

Reply:
Ahh, got it. Go does require a lock for concurrent read and write on map access.  Updating.